javagen/src/main/java/com/azure/autorest/Javagen.java

```diff
@@ -139,11 +148,35 @@ protected NodeTuple representJavaBeanProperty(Object javaBean, Property property
             }
 
             // TODO: POM, Manager
-            //Step 4: Print to files
+            //Step 4: format files
             Formatter formatter = new Formatter();
-            for (JavaFile javaFile : javaPackage.getJavaFiles()) {
-                String formattedSource = formatter.formatSourceAndFixImports(javaFile.getContents().toString());
-                writeFile(javaFile.getFilePath(), formattedSource, null);
+            Map<String, String> formattedFiles = javaPackage.getJavaFiles().stream().collect(Collectors.toMap(JavaFile::getFilePath, f -> {
+                try {
+                    return formatter.formatSourceAndFixImports(f.getContents().toString());
+                } catch (FormatterException e) {
+                    throw new RuntimeException(e);
+                }
+            }));
+            //Step 5: post processing
+            String jarPath = JavaSettings.getInstance().getPostProcessorJarPath();
```

Comment:
Not sure whether it post-process better be done before or after formatting.

It seems better as "before".
1. formatting might do hard-to-predict formatting that might affect simple replacing task (e.g. `this.getFoo()` get line break at `.` due to certain line limit)
2. seems better to reformat after the post-process

Reply:
Yes, agree with @weidongxu-microsoft  that we should format after post-processing. The formatter also catches some syntax errors which is useful to do after the postprocessing.

Reply:
But how do I know what strings to replace from the non-formatted code? The expected user scenario is they look at the formatted output code and they want to replace something. But that may not match the non-formatted code.

One way to work around it is to format the code again after post processing.

Also, we talked to C# folks and we are planning to not allow users do string replacements in the post processor - instead we will expose a few methods to do a set of "pre-approved" post processings, like "addCredentialToSyncAsyncClient", etc.

Reply:
OK. If the use case is user check the output then do the change, then it is reasonable to do post-process here.